crates/ruff_python_formatter/src/comments/placement.rs

```diff
@@ -267,17 +263,51 @@ fn handle_in_between_bodies_own_line_comment<'a>(
         return CommentPlacement::Default(comment);
     }
 
-    // it now depends on the indentation level of the comment if it is a leading comment for e.g.
-    // the following `elif` or indeed a trailing comment of the previous body's last statement.
+    // Check if we're between bodies and should attach to the following body. If that is not the
+    // case, either because there is no following branch or because the indentation is too deep,
+    // attach to the recursively last statement in the preceding body with the matching indentation.
+    match handle_own_line_comment_between_branches(comment, locator, preceding) {
+        CommentPlacement::Default(comment) => {
+            // Knowing the comment is not between branches, handle comments after the last branch
+            handle_own_line_comment_after_branch(comment, locator, preceding)
+        }
+        placement => placement,
+    }
```

Comment:
this could likely also be merged to avoid recomputing indents but i stopped the yak shave there.